cmd/tempo/app/modules.go

```diff
@@ -219,6 +219,7 @@ func (t *App) initDistributor() (services.Service, error) {
 func (t *App) initIngester() (services.Service, error) {
 	t.cfg.Ingester.LifecyclerConfig.ListenPort = t.cfg.Server.GRPCListenPort
 	t.cfg.Ingester.AutocompleteFilteringEnabled = t.cfg.AutocompleteFilteringEnabled
+	t.cfg.Ingester.DedicatedColumns = t.cfg.StorageConfig.Trace.Block.DedicatedColumns
```

Comment:
Ingesters don't have access to the storage config, but rather use `store.Store` which is an interface. Passing the config in here keeps interfaces clean.

Reply:
Yup. This is fine. There's just a number of places where we do this copying from one config to another and I often find it confusing. 

Since this is not settable by the user on the ingester config:

```
	DedicatedColumns             backend.DedicatedColumns `yaml:"-"`
```

Then I'm good.